pkg/authenticator/authenticator.go

```diff
@@ -233,3 +243,40 @@ func resetIAMURL(auth Authenticator) bool {
 	}
 	return false
 }
+
+// retry ...
+func retry(logger *zap.Logger, retryfunc func() error) error {
+
```

Comment:
Could you please add  a TODO here for making maxretry attempts/gap as configurable.

Reply:
Done